api/src/main/java/org/eclipse/microprofile/reactive/streams/CompletionSubscriber.java

```diff
@@ -51,7 +51,8 @@
 
     /**
      * Create a {@link CompletionSubscriber} by combining the given subscriber and completion stage.
-     *
+     * The objects passed to this method should not be reused.
```

Comment:
I think the more important thing to point out here on the javadoc for this class is that this subscriber, like all Reactive Streams subscribers is good for only one use.

Also, `CompletionStage's` are reusable, you can attach callbacks to them as much as you want, both before and after they are redeemed. It's just the subscriber that we need to be sure about.

Reply:
Hmmm, all the methods of CompletionStage that attach callbacks will return a new completion stage as a result the completion stage itself is 'sort of' immutable in terms of what it represents. 
I was trying to express not reusable in terms of that which they represent i.e. the completion of a single stream. For example reusing a CompletionStage as a parameter in calling this method more than once with the same completion stage would be bad. I will try to reword better in the review comment handling commit.

Reply:
Yes, a `CompletionStage` is "sort of" immutable, in contrast to a `Subscriber` where its state can and will change with any invocation of its callbacks.

> For example reusing a CompletionStage as a parameter in calling this method more than once with the same completion stage would be bad.

Not necessarily. In future, we could have a dynamic fan in feature, where you can create lots of `CompletionSubscriber`'s, and attach them to `Publisher`'s as needed. These could well all share the same `CompletionStage`, which might indicate when the subscribing stream is complete. Also, there could be circumstances where you might pass a subscriber that immediately cancels, and a shared `CompletionStage` this is already redeemed (or will be redeemed in the future), to multiple places, in cases where a `CompletionSubscriber` is needed by an API but the consumer of the API wants to subvert the stream and produce something different. There's no reason why that wouldn't be a valid thing to do, from the APIs perspective.

Reply:
Thank you, I enjoyed reading this carefully, both times :-).